torch/csrc/api/include/torch/nn/parallel/data_parallel.h

```diff
@@ -40,6 +150,11 @@ std::vector<std::shared_ptr<ModuleType>> replicate(
     replicas.push_back(
         std::dynamic_pointer_cast<ModuleType>(module->clone(device)));
   }
+  // Configure gradient edges to point from replcia parameters to original
+  // module parameters. See [Replicating Modules]
+  std::vector<std::shared_ptr<Module>> untyped_replicas(
+    replicas.begin(), replicas.end());
```

Comment:
I don't see why you have to allocate a vector here. Can't you just pass the iterators to the function?

Reply:
I guess this is because of the way you handle children, because you concatenate the replicas of all sibling children (not just the current child) and now you don't have iterators. But that seems like a pretty unusual to write the recursion: normally you'd just recurse on child modules.  Is there a reason it was done this way?

Reply:
I remember I run into problems when collecting children modules on ling 126. Let me try if I can drop this `untyped_replicas `.

Reply:
I think I can recurse on child modules by creating a map from `*parameter` to `grad_fn`, and retrieve that `grad_fn` in every recursion.